886 Print network profiles

adamfilipow92 opened the pull request:
Fixes #886 

## Test Plan
> How do we know the code works?

Command 

```

flank firebase test network-profiles list

```
and

```

flank network-profiles list

```

Should print table

```

┌────────────┬──────┬────────┬────────────┬───────────────────┬───────────┬───────┐
│ PROFILE_ID │ RULE │ DELAY  │ LOSS_RATIO │ DUPLICATION_RATIO │ BANDWIDTH │ BURST │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.040s │   0.001    │                   │  16000.0  │       │
│    LTE     │ down │ 0.040s │   0.001    │                   │  16000.0  │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.060s │   0.0015   │                   │  4000.0   │       │
│  LTE-poor  │ down │ 0.060s │   0.0015   │                   │  4000.0   │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.060s │   0.001    │                   │  2000.0   │       │
│    HSPA    │ down │ 0.060s │   0.001    │                   │  4000.0   │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.100s │   0.0015   │                   │   500.0   │       │
│ HSPA-poor  │ down │ 0.100s │   0.0015   │                   │  1000.0   │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.060s │    0.02    │                   │  2000.0   │       │
│    UMTS    │ down │ 0.060s │    0.02    │                   │  2000.0   │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.090s │    0.04    │                   │   384.0   │       │
│ UMTS-poor  │ down │ 0.090s │    0.04    │                   │   384.0   │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.150s │    0.04    │                   │   384.0   │       │
│    EDGE    │ down │ 0.150s │    0.04    │                   │   384.0   │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.200s │    0.08    │                   │   96.0    │       │
│ EDGE-poor  │ down │ 0.200s │    0.08    │                   │   96.0    │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.300s │    0.08    │                   │   172.0   │       │
│    GPRS    │ down │ 0.300s │    0.08    │                   │   172.0   │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.500s │    0.1     │                   │   48.0    │       │
│ GPRS-poor  │ down │ 0.500s │    0.1     │                   │   48.0    │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.500s │    0.08    │                   │    9.6    │       │
│    GSM     │ down │ 0.500s │    0.08    │                   │    9.6    │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.600s │    0.1     │                   │    2.4    │       │
│  GSM-poor  │ down │ 0.600s │    0.1     │                   │    2.4    │       │
└────────────┴──────┴────────┴────────────┴───────────────────┴───────────┴───────┘

```

also

```

flank firebase test android|ios test-environment 

```
should print this table at the end of the output. 

## WDYT

I added row separator to all tables to keep one style. In my opinion, it looks fine but on the other hand, the table with row separator takes more space. Tell me please what you think. Maybe we shouldn't add row separators to all tables.

## Checklist

- [X] Unit tested
- [X] release_notes.md updated
- [X] Printing network profiles on `flank firebase test network-profiles list`
- [X] Printing network profiles on `flank firebase test android|ios test-environment `

piotradamczyk5 commented on test_runner/src/test/kotlin/ftl/cli/firebase/test/networkprofiles/NetworkProfilesListCommandTest.kt:
```diff
@@ -33,19 +32,18 @@ class NetworkProfilesListCommandTest {
+            } returns configurationsMock
+            every { configurationsMock.asPrintableTable() } returns ""
+            CommandLine(NetworkProfilesListCommand()).execute()
+            verify { configurationsMock.asPrintableTable() }
```
I think that we have response for this call-in mocked server. Could we use it somehow?

adamfilipow92 replied:
Yes, we have calls to TestEnvironmentCatalog I change the test to using a mock server. Thanks!

adamfilipow92 replied:
I'm not 100% sure but probably I need to mock it to check printing is executed on command run.

piotradamczyk5 commented:
Could you add also the possibility to run table using a command
`flank network-profiles list`
?

piotradamczyk5 commented on test_runner/src/main/kotlin/ftl/cli/firebase/test/android/AndroidTestEnvironmentCommand.kt:
```diff
@@ -24,6 +26,7 @@ class AndroidTestEnvironmentCommand : Runnable {
         println(devicesCatalogAsTable(AndroidArgs.load(Paths.get(configPath)).project))
         println(supportedVersionsAsTable(AndroidArgs.load(Paths.get(configPath)).project))
         println(providedSoftwareAsTable())
+        println(GcTesting.networkConfiguration().asPrintableTable())
```
Could you update description to :
```
    header = ["Print available devices, OS versions, provided software list and network configuration to test against"],
    description = ["Print available Android devices, Android OS versions list, provided software and network configuration to test against"],
```

adamfilipow92 replied:
Changed

piotradamczyk5 commented on test_runner/src/main/kotlin/ftl/cli/firebase/test/ios/IosTestEnvironmentCommand.kt:
```diff
@@ -24,6 +26,7 @@ class IosTestEnvironmentCommand : Runnable {
         println(devicesCatalogAsTable(IosArgs.load(Paths.get(configPath)).project))
         println(softwareVersionsAsTable(IosArgs.load(Paths.get(configPath)).project))
         println(providedSoftwareAsTable())
+        println(GcTesting.networkConfiguration().asPrintableTable())
```
Could you update description to :
```
    header = ["Print available devices, OS versions, provided software list and network configuration to test against"],
    description = ["Print available iOS devices, iOS OS versions list, provided software and network configuration to test against"],
```
?

adamfilipow92 replied:
Changed

piotradamczyk5 commented:
```
WDYT
I added row separator to all tables to keep one style. In my opinion, it looks fine but on the other hand, the table with row separator takes more space. Tell me please what you think. Maybe we shouldn't add row separators to all tables.
```

For me defaults separator should not be printed cause tables become larger (readability is a plus, but the size is a minus :( )

adamfilipow92 commented:
> ```
> WDYT
> I added row separator to all tables to keep one style. In my opinion, it looks fine but on the other hand, the table with row separator takes more space. Tell me please what you think. Maybe we shouldn't add row separators to all tables.
> ```
> 
> For me defaults separator should not be printed cause tables become larger (readability is a plus, but the size is a minus :( )

So we thinking the same. I added a table style option and set default style without rows separation. Thanks for your feedback!

piotradamczyk5 commented:
Great work 👍
